setup.sh

```diff
@@ -41,59 +41,60 @@ declare -r DOCUMENTATION='documentation'
+}
+
 function check_dependencies()
 {
   local package_list=''
   local pip_package_list=''
   local cmd=''
   local distro
-
+  declare -a debian=('debian_pkg_check ' 'apt install -y')
+  declare -a arch=('pacman -Ql ' 'pacman -S')
+  declare -a gentoo=('qlist -I ' 'emerge')
+  declare -a fedora=('rpm -q ' 'dnf install -y')
+  declare -A distro_commands=(['arch']=arch ['gentoo']=gentoo ['fedora']=fedora ['debian']=debian)
+  declare -n distro_cmd
   distro=$(detect_distro '/')
 
-  if [[ "$distro" =~ 'arch' ]]; then
-    while IFS='' read -r package; do
-      installed=$(pacman -Ql "$package" &> /dev/null)
-      [[ "$?" != 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/arch.dependencies"
-    cmd="pacman -S $package_list"
-  elif [[ "$distro" =~ 'debian' ]]; then
-    while IFS='' read -r package; do
-      installed=$(dpkg-query -W --showformat='${Status}\n' "$package" 2> /dev/null | grep -c 'ok installed')
-      [[ "$installed" -eq 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/debian.dependencies"
-    cmd="apt install -y $package_list"
-  elif [[ "$distro" =~ 'fedora' ]]; then
+  distro_cmd="${distro_commands["$distro"]}"
+  distro_package_manager_cmd="${distro_cmd[1]}"
+  distro_package_check="${distro_cmd[0]}"
+
+  if [[ -v distro_commands["$distro"] ]]; then
     while IFS='' read -r package; do
-      installed=$(rpm -q "$package" &> /dev/null)
+      installed=$($distro_package_check "$package" &> /dev/null)
       [[ "$?" -ne 0 ]] && package_list="$package $package_list"
-    done < "$DOCUMENTATION/dependencies/fedora.dependencies"
-    cmd="dnf install -y $package_list"
-  else
-    warning 'Unfortunately, we do not have official support for your distro (yet)'
-    warning 'Please, try to find the following packages:'
-    warning "$(cat "$DOCUMENTATION/dependencies/arch.dependencies")"
-    return 0
-  fi
-
-  if [[ -n "$package_list" ]]; then
-    if [[ "$FORCE" == 0 ]]; then
-      if [[ $(ask_yN "Can we install the following dependencies $package_list?") =~ '0' ]]; then
-        return 0
+    done < "${DOCUMENTATION}/dependencies/${distro}.dependencies"
+    cmd="$distro_package_manager_cmd $package_list"
+
+    if [[ -n "$package_list" ]]; then
+      if [[ "$FORCE" == 0 ]]; then
+        if [[ $(ask_yN "Can we install the following dependencies $package_list?") =~ '0' ]]; then
+          return 0
+        fi
       fi
-    fi
 
-    # Install system packages
-    if [[ "$EUID" -eq 0 ]]; then
-      eval "$cmd"
-    else
-      eval "sudo $cmd"
+      # Install system packages
+      if [[ "$EUID" -eq 0 ]]; then
+        eval "$cmd"
+      else
+        eval "sudo $cmd"
+      fi
     fi
+  else
+    warning 'Unfortunately, we do not have official support for your distro (yet)'
+    warning 'Please, try to find the following packages:'
+    warning "$(cat "${DOCUMENTATION}/dependencies/arch.dependencies")"
```

Comment:
Use: https://kworkflow.org/content/codingstyle.html#command-substitution-and-arithmetic-expression

Reply:
okay